contracts/phoenix-bonds/src/token_receiver.rs

```diff
@@ -0,0 +1,58 @@
+use crate::*;
+use near_contract_standards::fungible_token::receiver::FungibleTokenReceiver;
+use near_sdk::{near_bindgen, PromiseOrValue};
+
+const MINIMUM_BOND_LINEAR_AMOUNT: u128 = ONE_NEAR / 10; // 0.1 LiNEAR
+const BOND_MSG: &str = "Bond";
+
+const ERR_BAD_MSG: &str = "Unrecognized message";
+const ERR_BAD_TOKEN: &str = "Only LiNEAR token can be used to bond";
+const ERR_SMALL_BOND_LINEAR_AMOUNT: &str = "Bond amount must be at least 0.1 LiNEAR";
+
+#[near_bindgen]
+impl FungibleTokenReceiver for PhoenixBonds {
+    #[allow(unused_variables)]
+    fn ft_on_transfer(
+        &mut self,
+        sender_id: AccountId,
+        amount: U128,
+        msg: String,
+    ) -> PromiseOrValue<U128> {
+        require!(env::prepaid_gas() >= GAS_FT_ON_TRANSFER, ERR_NOT_ENOUGH_GAS);
+        require!(msg == BOND_MSG, ERR_BAD_MSG);
```

Comment:
Suggest using `enum` type to represent msg rather than just a `str`. e.g.
```
#[derive(Deserialize)]
enum ReceiverMsg {
    Bond
}
```

Reply:
kk